src/dictionary.c

```diff
@@ -7,13 +7,16 @@
 #include "ptype2.h"
 #include "utils.h"
 
+#include "decl/dictionary-decl.h"
+
 // Initialised at load time
 struct vctrs_arg args_needles;
 struct vctrs_arg args_haystack;
 
 
 // http://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
-int32_t ceil2(int32_t x) {
+static inline
+uint32_t u32_ceil2(uint32_t x) {
```

Comment:
An overflow check might also be nice here, i.e. `x != 0` before returning. This will overflow when `x` is greater than `2^31`. Should not happen with short vector sizes but better be defensive?

Reply:
Oh neat!

I've added the check and a few comments about it

I've also ensured that if the input was `0`, it returns `2^0 = 1`. Previously it always returned `0` for that case, but that isn't a power of 2 and would be caught by the error check.

Reply:
b1f2354356b8a7cd31f2084cc9d1e9c3591fc241